pkg/test/step.go

```diff
@@ -422,6 +454,7 @@ func (s *Step) Check(namespace string, timeout int) []error {
 
 	if s.Assert != nil {
 		testErrors = append(testErrors, s.CheckAssertCommands(context.TODO(), namespace, s.Assert.Commands, timeout)...)
+		testErrors = append(testErrors, s.CheckAssertExpressions(context.TODO(), s.Assert.ResourceRefs, s.Assert.AssertAny, s.Assert.AssertAll)...)
```

Comment:
I don't know why `Step.CheckAssertCommands()` takes `Step.Assert.Commands` as arguments instead of just accessing it directly in the body, but let's not copy this pattern here.

However you will likely need the `namespace` argument here, otherwise `CheckAssertExpressions` will not know which ephemeral namespace to retrieve resources from!

Reply:
> However you will likely need the namespace argument here, otherwise CheckAssertExpressions will not know which ephemeral namespace to retrieve resources from!

`ResourceRefs` allow specifying the namespace right, so we can't be restricted to just the ephemeral `namespace`. WDYT?

Reply:
The namespace in ResourceRefs should probably take precedence such that it's possible to assert things in other namespaces (statically named). But the user cannot foresee what ephemeral namespace name kuttl will come up with, so for assertions on objects in the current test's namespace (ResourceRef would not specify one in that case), we need to propagate it here, similar to how it's fed to `CheckAssertCommands`.

Reply:
This is now incorporated in the latest commits. Please take a look when you have time. Thanks!